manager/volume.go

```diff
@@ -614,7 +617,59 @@ func (m *VolumeManager) Expand(volumeName string, size int64) (v *longhorn.Volum
 	}
 
 	if v.Spec.Size > size {
-		return nil, fmt.Errorf("cannot expand volume %v with current size %v to a smaller size %v", v.Name, v.Spec.Size, size)
+		return nil, fmt.Errorf("longhorn volume %v spec size %v is greater than requested Persistent Volume Claim size %v", v.Name, v.Spec.Size, size)
+
+	}
+
+	if v.Status.ActualSize > size {
+		return nil, fmt.Errorf("longhorn volume %v actual size %v is greater than requested Persistent Volume Claim size %v", v.Name, v.Spec.Size, size)
+	}
+
+	kubernetesStatus := &v.Status.KubernetesStatus
+	if kubernetesStatus.PVCName != "" && kubernetesStatus.LastPVCRefAt == "" {
+		pvc, err := m.ds.GetPersistentVolumeClaim(kubernetesStatus.Namespace, kubernetesStatus.PVCName)
+		if err != nil {
+			return nil, err
+		}
+
+		requestedSize := resource.MustParse(strconv.FormatInt(size, 10))
+
+		// TODO: Should check for pvc.Spec.Resources.Requests.Storage() here, once upgrade API to v0.18.x.
+		storageSpecValue, ok := pvc.Spec.Resources.Requests[corev1.ResourceStorage]
+		if !ok {
+			return nil, fmt.Errorf("cannot get request storage")
+		}
+		// TODO: Should check for pvc.Status.CapacityStorage() here, once upgrade API to v0.18.x.
+		storageStatusValue, ok := pvc.Status.Capacity[corev1.ResourceStorage]
+		if !ok {
+			return nil, fmt.Errorf("cannot get capacity storage")
+		}
+
+		if storageSpecValue.Cmp(requestedSize) == 0 && storageStatusValue.Cmp(requestedSize) == 0 && v.Spec.Size == size {
```

Comment:
Should we just check `storageSpecValue.Cmp(requestedSize) == 0`?

Reply:
End User can actually do this to LHV and it does not reflect in PV and PVC.
```
LHV=test
kubectl patch -p '{"spec":{"size": "3Gi"}}' volume.longhorn.io ${LHV} --namespace longhorn-system --type=merge
```

Reply:
If the user changes the size of LHV, it should be reconciled by the controller instead of API, right? Here, it should be simple, if users request the API (by UI), we just respect the requested value. After that, all things should be handled by the CSI resizer. 

https://github.com/longhorn/longhorn-manager/blob/713e2e48b6f7fe191ccd36d33b41fb2e5a234811/controller/volume_controller.go#L1850-L1850